src/Compilers/VisualBasic/Portable/Symbols/Attributes/PEAttributeData.vb

```diff
@@ -120,7 +120,7 @@ Namespace Microsoft.CodeAnalysis.VisualBasic.Symbols.Metadata.PE
 
         Private Sub EnsureLazyMembersAreLoaded()
 
-            If _lazyConstructorArguments Is Nothing Then
+            If Volatile.Read(_lazyConstructorArguments) Is Nothing Then
```

Comment:
>Nothing [](http://example.com/codeflow?start=59&length=7)

Do we need `OrElse Volatile.Read(_lazyNamedArguments) Is Nothing` here as well, or `Volatile.Read(_lazyNamedArguments) Is Nothing` in `CommonNamedArguments`?

Reply:
I don't see anything in this type that is gated on it. Since `EnsureLazyMembers` uses `_lazyConstructorArguments` as the gate for whether to initialize everything, and assigns that last, the access in `CommonNamedArguments` should be safe: either `_lazyNamedArguments` was non-default, and the second read is fine, or it wasn't null, and this `Volatile.Read` will ensure that the second read of `_lazyNamedArguments` isn't reordered before a potential initialization.